### ISSUE TITLE
Vision data quality analysis

### ISSUE THREAD

**Issue by aniketmaurya** [PyTorchLightning/lightning-flash#194]
## 🚀 Feature
Inbuilt method to check data where model fails by high margin.

### Motivation
Easier error analysis and model interpretation

### Pitch
Most of the time data quality is responsible for model performance.
When I train a vision model I calculate loss for each image in my dataset and sort it by highest loss. Most of the time I find that the data is wrong labelled or too noisy.
It would be great to have an inbuilt method in flash to get data points where model fails to predict correctly by high margin.

May be we can have some get_top_wrong_preds method that we can call and get those data. 

### Additional context

As end user my main expectation is to easily find out why my model is not learning or generalising.
One more feature can be to provide inbuilt support for model interpretation (using Captum may be). For example, we can have some easy to use function for GradCAM interpretation inside ImageClassificationTask
Reference to one of my small implementation in TF 2: https://github.com/aniketmaurya/chitra#model-visualization

**Comment by addisonklinke** [comment]
A principled approach for finding label errors in datasets is discussed in this [blog post](https://l7.curtisnorthcutt.com/confident-learning) referencing the [paper](https://arxiv.org/abs/1911.00068) "Confident Learning: Estimating Uncertainty in Dataset Labels". The authors have already published a Python package [`cleanlab`](https://github.com/cgnorthcutt/cleanlab) which according to the Github readme

> is python package for machine learning with noisy labels. cleanlab cleans labels and supports finding, quantifying, and learning with label errors in datasets

This seems to be an excellent candidate to do most of the heavy lifting required for a PR that addresses this feature request

**Comment by edgarriba** [comment]
I think it's also worth to look at https://github.com/torchdrift/torchdrif /cc @t-vi
